src/lib/dialog/dialog-content-directives.ts

```diff
@@ -56,6 +62,9 @@ export class MatDialogClose implements OnInit, OnChanges {
       // be resolved at constructor time.
       this.dialogRef = getClosestDialog(this._elementRef, this._dialog.openDialogs)!;
     }
+
+    const buttonText = this._elementRef.nativeElement.innerText;
+    this._hasText = !!buttonText && buttonText.trim().length > 0;
```

Comment:
A note on this: this could technically become inaccurate if the button has a data binding that starts off with a value, but then gets cleared. I decided not to handle that case, because of the potential performance implications.